### PR TITLE
Add a deprecation helper to detect insert tags in Twig templates

### PR DESCRIPTION
**Pull request by m-vo**
This PR introduces a Twig node visitor that adds `Twig\Node\DeprecatedNode` nodes when a specific usage is found during compilation.

The first implemented scenario detects this pattern `{{ '{{an_insert_tag'}}' }}` and suggests using the `insert_tag` function instead:
image

The analysis only happens at compile time and does not affect runtime performance.

/cc @ausi

### CODE REVIEW

**Comment by ausi** [comment]
We could maybe even go as far as checking if there is a node that starts with `{{ '{{…` so every print node that starts with a string literal that starts with `{{`.

**Comment by m-vo** [comment]
> We could maybe even go as far as checking if there is a node that starts with {{ '{{… so every print node that starts with a string literal that starts with {{.

This gets complicated quickly, because then generating suggestions (or even displaying the usage) must be done at runtime. Because there could be any sort of (nested) nodes in the print node, we would need to execute and capture the output (same as with the `add` tag). But I would avoid doing this as it could easily introduce sideeffects.

**Review by ausi — DISMISSED** [review]

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @m-vo.

File: core-bundle/src/Twig/Extension/DeprecationsNodeVisitor.php [hunk 1]
```diff
@@ -57,19 +57,19 @@ private function handleDeprecatedInsertTagUsage(Node $node): Node
...
     }
 
-    private function addDeprecation(Node $node, string $version, string $message): Node
+    private function addDeprecation(Node $node, string $message): Node
     {
         $line = $node->getTemplateLine();
 
         $deprecatedNode = new DeprecatedNode(
-            new ConstantExpression("Since contao/core-bundle $version: $message", $line),
+            new ConstantExpression("Since contao/core-bundle 4.13: $message", $line),
```
Comment (m-vo) [inline core-bundle/src/Twig/Extension/DeprecationsNodeVisitor.php:72]: @leofeyer Why did you remove the version variable?

Comment (leofeyer) [reply core-bundle/src/Twig/Extension/DeprecationsNodeVisitor.php:72]: Because it is always `4.13` and thus we don‘t need a separate variable.

Comment (m-vo) [reply core-bundle/src/Twig/Extension/DeprecationsNodeVisitor.php:72]: The idea was that we could have more handlers in the future (e.g. `handleDeprecatedFooBarUsage`) and they could add deprecations similar to `trigger_deprecation`.

Comment (leofeyer) [reply core-bundle/src/Twig/Extension/DeprecationsNodeVisitor.php:72]: Sure. We can still add a variable then.
